Various enhancements and improvements

verpeteren opened the pull request.

verpeteren committed:
Changed formatting style from Mozilla to file

verpeteren committed:
Moved files around

verpeteren committed:
Renamed output directory to cpp

verpeteren committed:
Moved idl file into idl directory

verpeteren committed:
Preparations for multiple file creation and easier setup.

verpeteren committed:
Automatically process all idl input files.

verpeteren committed:
Renamed example/Hello to example/Demo

verpeteren committed:
Take the correct classname to generate the file and handle constructor arguments.

verpeteren committed:
Add another demo (Fragment)

verpeteren committed:
Renamed 'base_*' templates to 'impl_*'

verpeteren committed:
Generate the cpp code based on the template.

verpeteren committed:
style

verpeteren committed:
Transferred research from NidiumTools:idl_classmapper branch.

verpeteren committed:
Added the NidiumModule example (WIP)

verpeteren committed:
Added the Nidium example (WIP)

verpeteren committed:
Added the automatically generated idf files for Nidium.

Please note: this is just temporary. These files should/will be removed soon.

verpeteren committed:
Added more details to the README

verpeteren committed:
Added LICENSE texts

verpeteren committed:
Added Prefix to classname.

verpeteren committed:
NidiumModule 'Hello' looks good.

verpeteren committed:
Corrected several inconsistancies for static methods.

verpeteren committed:
Registration process works with the Axtended Attribute 'exposed':

5 flavors
* not set: not exposed
* 'class': build in (Socket)
* 'embed': build, but not loaded in the namespace by default (DB)
* 'module': module that is dynamically loaded
* /a-z/+: build in module, that is loaded only once, and it's instance is put in the global object.

todo: Alias & Implemented as

efyx commented:
I'd like to merge this into the master branch as it brings several improvements and is more up to date.

@paraboul any objections ?

verpeteren committed:
Processed feedback

- rename class <name>Base
- indentation
- removed ctor/dtor
- fixed a bug in the registration
- removed '->root()'
- added a 'void registrationHook()'
